service/src/ocall_bridge/sidechain_ocall.rs

```diff
@@ -67,21 +67,22 @@ where
 	PeerBlockFetcher: FetchBlocksFromPeer<SignedBlockType = SignedSidechainBlock>,
 	TokioHandle: GetTokioHandle,
 {
-	fn propose_sidechain_blocks(&self, signed_blocks_encoded: Vec<u8>) -> OCallBridgeResult<()> {
+	fn propose_sidechain_blocks(&self, signed_blocks_encoded: &[u8]) -> OCallBridgeResult<()> {
 		// TODO: improve error handling, using a mut status is not good design?
 		let mut status: OCallBridgeResult<()> = Ok(());
 
+		let mut slice = signed_blocks_encoded;
```

Comment:
This is the "trick": Create a mutable copy of the slice (as `decode()` needs it). Maybe this could be written nicer?